src/main/java/org/ethereum/beacon/discovery/storage/LocalNodeRecordStore.java

```diff
@@ -9,26 +9,37 @@
 import org.ethereum.beacon.discovery.schema.NodeRecord;
 
 public class LocalNodeRecordStore {
+
   private volatile NodeRecord latestRecord;
   private final Bytes privateKey;
+  private final NodeRecordListener recordListener;
 
   public LocalNodeRecordStore(final NodeRecord record, final Bytes privateKey) {
-    latestRecord = record;
+    this(record, privateKey, (a, b) -> {});
+  }
+
+  public LocalNodeRecordStore(
+      NodeRecord record, Bytes privateKey, NodeRecordListener recordListener) {
+    this.latestRecord = record;
     this.privateKey = privateKey;
+    this.recordListener = recordListener;
   }
 
   public NodeRecord getLocalNodeRecord() {
     return latestRecord;
   }
 
   public void onSocketAddressChanged(final InetSocketAddress newAddress) {
```

Comment:
Probably worth some quick unit tests to verify the listener is invoked as expected.